File: razer-sdk/src/test/java/com/bonepl/razersdk/animation/FrameTest.java
package com.bonepl.razersdk.animation;

import com.bonepl.razersdk.sdk.RzKey;
import org.junit.jupiter.api.Test;

import java.util.EnumMap;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertEquals;

class FrameTest {
    @Test
    void testCustomEffectCreation() {
        //given
        final EnumMap<RzKey, Color> inputMap = new EnumMap<>(RzKey.class);
        inputMap.put(RzKey.RZKEY_ESC, Color.BLUE);
        inputMap.put(RzKey.RZKEY_ENTER, Color.WHITE);

        //when
        final Frame frame = new Frame(inputMap);
        final Map<RzKey, Color> actualMap = frame.getKeysToColors();

        //then
        assertEquals(inputMap, actualMap);
    }
}

File: test/razer-sdk/animation/Frame.test.ts
import {describe, expect, test} from 'vitest';
import {Frame} from '../../../src/razer-sdk/animation/Frame.js';
import {StaticColor} from '../../../src/razer-sdk/color/StaticColor.js';
import {Color} from '../../../src/razer-sdk/color/Color.js';
import {RzKey} from '../../../src/razer-sdk/sdk/RzKey.js';

describe('FrameTest', () => {
    test('testCustomEffectCreation', () => {
        //given
        const inputMap = new Map<RzKey, Color>([
            [RzKey.RZKEY_ESC, StaticColor.BLUE],
            [RzKey.RZKEY_ENTER, StaticColor.WHITE],
        ]);

        //when
        const frame = new Frame(inputMap);
        const actualMap = frame.getKeysToColors();

        //then
        expect(actualMap.size).toBe(inputMap.size);
        expect(actualMap.get(RzKey.RZKEY_ESC)).toBe(StaticColor.BLUE);
        expect(actualMap.get(RzKey.RZKEY_ENTER)).toBe(StaticColor.WHITE);
    });
});
